### PR TITLE
Implement FindUsageProvider

### PR DESCRIPTION
**Pull request by Turbo87**
bildschirmfoto 2015-11-13 um 13 17 44

### CODE REVIEW

File: src/org/rust/lang/RustDescriptionProvider.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+class RustDescriptionProvider : ElementDescriptionProvider {
+    override fun getElementDescription(element: PsiElement, location: ElementDescriptionLocation): String? {
+        return when (location) {
+            is UsageViewNodeTextLocation -> getNodeText(element)
+            is UsageViewShortNameLocation -> getShortName(element)
+            is UsageViewLongNameLocation -> getLongName(element)
+            is UsageViewTypeLocation -> getType(element)
+            is HighlightUsagesDescriptionLocation -> getLongName(element)
+            else -> null
```
Comment (matklad) [inline src/org/rust/lang/RustDescriptionProvider.kt]: We are trying to keep arrow aligned in `when`. There is a checkbox for it in code style config.

File: src/org/rust/lang/core/lexer/RustTokenElementTypes.java [hunk 2]
```diff
@@ -148,4 +148,10 @@
         com.intellij.psi.TokenType.WHITE_SPACE
     );
 
+    TokenSet COMMENTS_TOKEN_SET = TokenSet.create(
```
Comment (matklad) [inline src/org/rust/lang/core/lexer/RustTokenElementTypes.java]: This duplicates `getCommentTokens` from `RustParserDefinition`

File: src/org/rust/lang/RustDescriptionProvider.kt (hunk 1 above)
Comment (Turbo87) [inline src/org/rust/lang/RustDescriptionProvider.kt]: damn, I actually have that checkbox ticked...

File: src/org/rust/lang/core/lexer/RustTokenElementTypes.java (hunk 2 above)
Comment (Turbo87) [inline src/org/rust/lang/core/lexer/RustTokenElementTypes.java]: Yes, but the parser definition is not useable without an instance. I'm can reuse this one in the parser definition though.

**Pull request merged**